### DISCUSSION TITLE
Why does resizing a black and white image add colour?

### DISCUSSION THREAD

**Discussion by adamscoble** [ImageMagick/ImageMagick#6991] [Help]
### ImageMagick version

7.1.1-25

### Operating system, version and so on

Windows 11

### Description

Hi there,

I'm resizing a black and white image, and the resulting image has a few pixels of noticeable colour. I've tried converting to gray, however this makes the image much less light. 

This is the source image:
![source](https://github.com/ImageMagick/ImageMagick/assets/3459709/51170f57-de65-49df-93b4-c9274db8816e)

And this is the resulting resize:
![result](https://github.com/ImageMagick/ImageMagick/assets/3459709/ab0103d7-738d-4774-a708-1b932617fa11)
![result_large](https://github.com/ImageMagick/ImageMagick/assets/3459709/392a1486-8383-4c43-b58b-376a60224f8b)

(Note the pink pixel on the foot.) The command I'm using is:

`magick source.png -colorspace rgb -interpolate nearest -interpolative-resize 25% -channel a -threshold 15% -background white -alpha remove -alpha off "result.png"`

The reason I use `-colorspace rgb` is because the actual grayscale values are important, e.g. the top of the head should be RGB~0.678. By using RGB colorspace I get the closest to this. Using `-colorspace gray` removes any colour, but also I get very different lightness values. I tried tweaking with `-gamma` but it didn't seem to actually do anything. I also tried `lineargray`, and `-fx "(r+g+b)/3"` but all of them yielded inaccurate gray values.

Thanks for any help!

**Comment by snibgo** [comment]
> I'm resizing a black and white image, 

The input image is not black and white. It contains colours. `-verbose info:` shows it has RGB channels that contain different values. If we convert to HCL, increase the chroma, and convert back to sRGB, we can see the colours more clearly:
```
magick silman.png -colorspace HCL -channel 1 -evaluate Multiply 100 +channel -colorspace sRGB x.png
```
![x](https://github.com/ImageMagick/ImageMagick/assets/33812368/f54044b6-7ccf-4a5e-8488-3442094fc890)

`-colorspace gray` makes the top of the head about 0.678 (on a scale of 0 to 1). To show this, we crop to one pixel and write the value:
```
C:\web\im>%IMG7%magick silman.png -colorspace gray -crop 1x1+120+36 txt:
# ImageMagick pixel enumeration: 1,1,0,65535,graya
0,0: (44130,65535)  #AC62AC62AC62FFFF  graya(67.3382%,1)
```

**Reply by adamscoble** [reply]
Thanks for the reply! 

> `-colorspace gray` makes the top of the head about 0.678 (on a scale of 0 to 1). To show this, we crop to one pixel and write the value:

Interesting! In Photoshop it shows the top of the head at about 0.737, is the issue that I'm testing in Photoshop then?

Edit: Just to confirm, do you mean the source image is .678 at the head? Or the resized image? Basically I want both to be .678!

**Comment by snibgo** [comment]
I named your input image `silman.png`. After converting it to gray, one pixel near the top of the head was 67.3382%.

Gimp gives the same result. I don't use Photoshop.

**Reply by adamscoble** [reply]
Hi there,

Here is the source image:

![source](https://github.com/ImageMagick/ImageMagick/assets/3459709/23fc9c72-bde1-4723-9b16-c462179e1c53)

And here is the result of calling:

`magick source.png -colorspace gray result.png`

![result](https://github.com/ImageMagick/ImageMagick/assets/3459709/496be6d5-0e0b-48b0-83cb-0d9ca40936d9)

The result is lighter than the source. I'm not sure why this happens? But a pixel near the top of the head is now ~0.74.

Apologies if I'm misunderstanding!

**Comment by snibgo** [comment]
For some reason, I can't download the two images from your last post. If you zip them and post that, I expect I will be able to read them.

Did you try my command that crops a single pixel? What is the result?
```
magick source.png -colorspace gray -crop 1x1+120+36 txt:

magick result.png -colorspace gray -crop 1x1+120+36 txt:
```

**Reply by adamscoble** [reply]
So I tried your command on both files and got:

```
magick source.png -crop 1x1+120+36 txt:
# ImageMagick pixel enumeration: 1,1,0,65535,srgba
0,0: (43947,44203,43947,65535)  #ABABACABABABFFFF  srgba(67.0588%,67.4495%,67.0588%,1)
```
```
magick result.png -crop 1x1+120+36 txt:
# ImageMagick pixel enumeration: 1,1,0,65535,graya
0,0: (44130,65535)  #AC62AC62AC62FFFF  graya(67.3381%,1)
```

So the colour is very slightly different, but still in the ballpark. But it's bizarre, because in Photoshop source's gray value near the top of the head is 172 and result's is 188. And result looks noticeably lighter than source!

Here's a zip of the two: [gray.zip](https://github.com/ImageMagick/ImageMagick/files/13814250/gray.zip)

Thanks very much for your help by the way!

**Comment by snibgo** [comment]
Your zipped files look fine. The values are as I expect. They don't look different to me, but I don't use Photoshop.

The files contain some Photoshop metadata, which we can see with `magick source.png xmp:`. We can strip that metadata:
```
magick source.png -colorspace gray -strip result.png
```
Does that help, when you view the result in Photoshop?

**Reply by adamscoble** [reply]
Sorry for the delayed response! Sadly stripping the Photoshop metadata doesn't seem to help either. Here are the two files overlaid:

![mix](https://github.com/ImageMagick/ImageMagick/assets/3459709/9ed9ceac-5d1e-4fa8-b4fe-93573b51b65f)

The lighter version is result.png. So if you overlay them in Gimp you don't see different values? Both are 8-bit PNGs, though the result file is grayscale. Source's gray value is ~173 at the head, result's is ~189. Very strange!

**Comment by snibgo** [comment]
I don't know why Photoshop changes the pixel values.

**Comment by fmw42** [comment]
What about spot colors or dot gain settings in Photoshop.

**Reply by adamscoble** [reply]
Hey, thanks for the reply, though I'm struggling to find these. Would settings like this change things on open?
